scenario/constraints/MigrationConstraint.ts

```diff
@@ -30,7 +25,12 @@ export class MigrationConstraint<T extends CometContext, R extends Requirements>
         // Make proposer the default signer
         ctx.deploymentManager._signers.unshift(proposer);
 
+        // Order migrations deterministically and store in the context (i.e. for verification)
         migrationList.sort((a, b) => a.name.localeCompare(b.name))
+        ctx.migrations = migrationList;
+
+        // XXX This should check that a migration has not already been run/proposed on-chain.
+        // Otherwise the scenario could be running the same proposal twice.
```

Comment:
Added this XXX because the rewards migration is still being run in the scenario even though it's already been created as a proposal on-chain. I have to comment out the `MigrationConstraint` and `VerifyMigrationConstraint` locally when running scenarios because the rewards migration is not idempotent.

Reply:
Do we want to add back the enacted flag? Maybe we write it out in enact automatically

Reply:
We will do this in a separate PR